ledger/participant-integration-api/src/main/scala/platform/store/cache/StateCache.scala

```diff
@@ -16,11 +16,16 @@ import scala.concurrent.{ExecutionContext, Future}
 /** This class is a wrapper around a Caffeine cache designed to handle
   * correct resolution of concurrent updates for the same key.
   */
-private[platform] case class StateCache[K, V](cache: Cache[K, V], registerUpdateTimer: Timer)(
-    implicit ec: ExecutionContext
-) {
+private[platform] case class StateCache[K, V](
+    initialCacheIndex: Long,
+    cache: Cache[K, V],
+    registerUpdateTimer: Timer,
+)(implicit ec: ExecutionContext) {
   private val logger: ContextualizedLogger = ContextualizedLogger.get(getClass)
   private[cache] val pendingUpdates = mutable.Map.empty[K, PendingUpdatesState]
```

Comment:
Seems like quite a large change at this stage but if rather than a `pendingUpdates` map there was a `lastUpdated` map then that could be used to synchronously check that the index is monotonically increasing just prior to cache update.  

Asynchronous fetches would note the index at fetch time and then pass that into the sychronous update method.

Reply:
Problem with that approach is that the `lastUpdated` cache would continually increase in size, `pendingUpdates` can be cleaned up on cache update.